src/frontend/src/eth/components/transactions/Transaction.svelte

```diff
@@ -2,43 +2,84 @@
+	import { isSupportedEthTokenId } from '$eth/utils/eth.utils';
 	import { isTransactionPending } from '$eth/utils/transactions.utils';
+	import IconBurn from '$lib/components/icons/IconBurn.svelte';
+	import IconConvert from '$lib/components/icons/IconConvert.svelte';
+	import IconMint from '$lib/components/icons/IconMint.svelte';
 	import IconReceive from '$lib/components/icons/IconReceive.svelte';
 	import IconSend from '$lib/components/icons/IconSend.svelte';
 	import TransactionPending from '$lib/components/transactions/TransactionPending.svelte';
 	import Amount from '$lib/components/ui/Amount.svelte';
 	import Card from '$lib/components/ui/Card.svelte';
 	import RoundedIcon from '$lib/components/ui/RoundedIcon.svelte';
-	import { ethAddress } from '$lib/derived/address.derived';
+	import { tokenId, tokenSymbol } from '$lib/derived/token.derived';
 	import { i18n } from '$lib/stores/i18n.store';
 	import { modalStore } from '$lib/stores/modal.store';
-	import type { Transaction } from '$lib/types/transaction';
+	import { token } from '$lib/stores/token.store';
 	import { formatSecondsToDate } from '$lib/utils/format.utils';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
 
-	export let transaction: Transaction;
+	export let transaction: EthTransactionUi;
 
-	let from: string;
 	let value: BigNumber;
 	let timestamp: number | undefined;
 	let displayTimestamp: number | undefined;
+	let type: EthTransactionType;
 
 	let pending: boolean;
 	$: pending = isTransactionPending(transaction);
 
-	$: ({ from, value, timestamp, displayTimestamp } = transaction);
+	$: ({ value, timestamp, displayTimestamp, uiType: type } = transaction);
 
-	let type: EthTransactionType;
-	$: type = from?.toLowerCase() === $ethAddress?.toLowerCase() ? 'send' : 'receive';
+	let ckTokenSymbol: string;
+	$: ckTokenSymbol =
+		nonNullish($tokenId) && isSupportedEthTokenId($tokenId)
+			? `ck${$tokenSymbol}`
+			: ($token as Erc20Token).twinTokenSymbol ?? '';
 
 	let label: string;
-	$: label = type === 'send' ? $i18n.send.text.send : $i18n.receive.text.receive;
+	$: label =
+		type === 'withdraw'
+			? replacePlaceholders(
+					pending
+						? $i18n.transaction.label.converting_ck_token
+						: $i18n.transaction.label.ck_token_converted,
+					{
+						$twinToken: $tokenSymbol ?? '',
+						$ckToken: ckTokenSymbol
+					}
+				)
+			: type === 'deposit'
+				? replacePlaceholders(
+						pending
+							? $i18n.transaction.label.converting_twin_token
+							: $i18n.transaction.label.ck_token_sent,
+						{
+							$twinToken: $tokenSymbol ?? '',
+							$ckToken: ckTokenSymbol
+						}
+					)
+				: type === 'send'
+					? $i18n.send.text.send
+					: $i18n.receive.text.receive;
 
 	let icon: ComponentType;
-	$: icon = type === 'send' ? IconSend : IconReceive;
+	$: icon =
+		(type === 'withdraw' || type === 'deposit') && pending
+			? IconConvert
+			: type === 'withdraw'
+				? IconMint
+				: type === 'deposit'
```

Comment:
It's correct, deposit = burn?

Reply:
yes, they represent the same "scope" in the ck-conversions, but I am using the naming of the [documentation](https://github.com/dfinity/ic/blob/master/rs/ethereum/cketh/docs/cketh.adoc) for the transaction types. For the icons, I was planning already to change the name of the icons in another PR, so that we call them in another way, representing this

Reply:
Thanks for the explanation.

Reply:
Changed icons name with https://github.com/dfinity/oisy-wallet/pull/2347